Rework compat bindings.

robieta opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #47865 Enable callgrind collection for C++ snippets
* #47864 Add support for timing C++ snippets.
* **#47863 Rework compat bindings.**
* #47862 Add option for cpp_extensions to compile standalone executable
* #47861 Expose CXX_FLAGS through __config__

Differential Revision: [D25199261](https://our.internmc.facebook.com/intern/diff/D25199261)

Commit:
Rework compat bindings.

[ghstack-poisoned]

ezyang commented on torch/utils/benchmark/utils/cpp_jit.py:
```diff
@@ -0,0 +1,44 @@
+    # load will automatically search /usr/include, but not conda include.
+    EXTRA_INCLUDE_PATHS.append(os.path.join(CONDA_PREFIX, "include"))
+
+
```
It would probably be nice to have a Note explaining at a high level what's going on here

ezyang replied:
I understand that this is "temporary" stuff but because it interacts nontrivially with some code that isn't in PyTorch itself, it will be harder for other people to figure out how this relates to the bigger picture. A note here will help a lot.

robieta replied:
No, it's a valid point. Check out `BACK_TESTING_NOTE:` and let me know if it seems reasonable.

Commit:
Update on "Rework compat bindings."




[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."




[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."




[ghstack-poisoned]

ezyang commented on torch/utils/benchmark/utils/cpp_jit.py:
```diff
@@ -0,0 +1,68 @@
+#   PyTorch the cost and complexity of such shims will increase. Once back
+#   testing is no longer required (which is to say we have done enough historic
+#   analysis and the shims no longer justify their maintenance and code
+#   complexity costs) back testing paths will be removed.
```
Thanks, this note is super helpful. One extra benefit of the note is it also clues the reader in on what kinds of code changes to Timer are permissible, and what are not (changes that add more dependencies on C symbols => you need to add more backtesting support.)

Commit:
Update on "Rework compat bindings."




[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."


Differential Revision: [D25199261](https://our.internmc.facebook.com/intern/diff/D25199261)

[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."


Differential Revision: [D25199261](https://our.internmc.facebook.com/intern/diff/D25199261)

[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."


Differential Revision: [D25199261](https://our.internmc.facebook.com/intern/diff/D25199261)

[ghstack-poisoned]

Commit:
Update on "Rework compat bindings."


Differential Revision: [D25199261](https://our.internmc.facebook.com/intern/diff/D25199261)

[ghstack-poisoned]
